### PR TITLE
Add SWI Prolog

### LINKED ISSUE #6110: Request: SWI Prolog
**Brand Name:** SWI Prolog

**Website:** https://www.swi-prolog.org/

**Alexa rank:** 338,137
  
**Similarweb rank:** 106,829

**Resources for icon and color:**
  

https://github.com/SWI-Prolog/plweb-www/issues/16

Full logo: https://www.swi-prolog.org/icons/vector/swipl-logo.svg

https://github.com/SWI-Prolog/plweb-www/blob/7e2bd72621c8634ae9d39bfd8ce805719c0cc839/icons/vector/swipl-logo.svg

Full logo without text: https://starbeamrainbowlabs.com/images/logos/swi-prolog.svg

Monochrome variation: https://seekicon.com/free-icon/prolog_1

Color: `#ed1c24` (from the [JetBrains Prolog extension logo](https://plugins.jetbrains.com/plugin/13954-prolog) which has flat colors)

### PR DESCRIPTION
**Pull request by DenverCoder1**
image

**Issue:** closes #6110

**Similarweb rank:** 296,571

### Checklist

- [x] I updated the JSON data in `_data/simple-icons.json`
- [x] I optimized the icon with SVGO or SVGOMG
- [x] The SVG `viewbox` is `0 0 24 24`

### Description

Full logo: https://www.swi-prolog.org/icons/vector/swipl-logo.svg
Full logo without text: https://starbeamrainbowlabs.com/images/logos/swi-prolog.svg
Monochrome: https://seekicon.com/free-icon/prolog_1

The original logo was used with minor modifications to make it look good as monochrome such as the outline around the beak.

Color: `#ed1c24` (from the [JetBrains Prolog extension logo](https://plugins.jetbrains.com/plugin/13954-prolog) which has flat colors)

### CODE REVIEW

**Comment by ThatOneCalculator** [comment]
> https://github.com/simple-icons/simple-icons/actions/runs/6427071279/job/17452170962?pr=9687

Should probably be above Swift (line 12979)

File: _data/simple-icons.json [hunk 1]
```diff
@@ -12975,6 +12975,11 @@
             "hex": "000000",
             "source": "https://github.com/swc-project/logo/blob/f26cac1b4a490e3bdf128d3b084bb57f4fab1aac/svg/swc_black.svg"
         },
+        {
+            "title": "SWI Prolog",
+            "hex": "ED1C24",
+            "source": "https://seekicon.com/free-icon/prolog_1"
```
Comment (PratyushJayachandran) [inline _data/simple-icons.json:12981]:
```suggestion
            "source": "https://www.swi-prolog.org/icons/vector/swipl-logo.svg"
```
We would prefer to use official source always when available.

File: icons/swiprolog.svg [hunk 2]
```diff
@@ -0,0 +1 @@
...27-2.127zm9.869.084a2.128 2.128 0 1 0 .002 4.256 2.128 2.128 0 0 0-.002-4.256zm-9.82.81c.714 0 1.287.577 1.287 1.29a1.28 1.28 0 0 1-1.286 1.283 1.286 1.286 0 1 1 0-2.573zm9.746.018c.712 0 1.283.577 1.283 1.289s-.576 1.285-1.283 1.285a1.288 1.288 0 1 1 0-2.574zm-10.325.445a.362.362 0 0 0-.361.36c0 .198.164.363.36.363.198 0 .363-.16.363-.362a.365.365 0 0 0-.362-.36Zm9.739.006a.362.362 0 0 0-.362.36c0 .198.165.36.362.36a.356.356 0 0 0 .359-.36.36.36 0 0 0-.36-.36zm-5.157 1.344a1.98 1.98 0 0 0-1.982 1.982c0 .877.62 1.63 1.004 2.37.164.32.332.64.496.958.145.277.3.585.482.862l.746-1.377c.249-.478.502-.956.75-1.434.244-.454.487-.85.487-1.383a1.98 1.98 0 0 0-1.983-1.978Zm-9.869 4.283.53.488.826-.29-.957 1.003Zm19.737.146-.396 1.2-.957-1.002.824.289.53-.487zm-17.825.127.492.53.844-.23-1.031.927Zm15.915.145-.31 1.228-1.03-.927.842.228.498-.53zm-14.083.22.493.53.843-.23-1.031.927-.305-1.226Zm12.246.145-.308 1.229-1.027-.928.843.228zM2.764 15.56l.529.486.824-.29-.955 1.003-.398-1.2zm18.524.144-.398 1.2-.957-1.003.824.29zm-16.692.238.529.489.824-.291-.951 1.004Zm14.86.14-.4 1.202-.954-1.004.824.29zM3.271 17.65l.532.486.824-.29-.957 1.003zm17.503.145-.398 1.2-.955-1.004.824.291.53-.487z"/></svg>
```
Comment (PratyushJayachandran) [inline icons/swiprolog.svg:1]: It doesn't match with the original. [See here!](https://github.com/simple-icons/simple-icons/assets/35000807/10fb07cc-0c5d-473e-9bc4-bbce4166881d)

**Review by PratyushJayachandran — CHANGES_REQUESTED** [review]

**Comment by DenverCoder1** [comment]
@PratyushJayachandran The latest commit now has a monochrome version I made myself using the original logo from the SWI Prolog website. In order to make it look better in monochrome, I made the beak smaller so it should line up with the original besides the beak.

[image](https://github.com/simple-icons/simple-icons/assets/20955511/2e4550f2-19ff-49e3-b9a2-3999e0a7acc8)

**Comment by PratyushJayachandran** [comment]
> @PratyushJayachandran The latest commit now has a monochrome version I made myself using the original logo from the SWI Prolog website. In order to make it look better in monochrome, I made the beak smaller so it should line up with the original besides the beak.
> 
> [image](https://github.com/simple-icons/simple-icons/assets/20955511/2e4550f2-19ff-49e3-b9a2-3999e0a7acc8)

Thank you for contributing and taking the effort to make the logo monochrome yourself. Unfortunately I still see some differences like this:
image
Could be happening because of scaling issues. Can you take another look. I hope you are using this as source: https://www.swi-prolog.org/icons/vector/swipl-logo.svg

**Comment by DenverCoder1** [comment]
Not sure why it would be different, could be from transforming before and after optimizations and rounding errors.

I made a new one following the same steps but without multiple exports and optimizations. Hopefully it is correct now.

It looks to me like it is resolved. [image](https://github.com/simple-icons/simple-icons/assets/20955511/ceb2ac66-6089-466a-a012-aa1fb9c9c799) [image](https://github.com/simple-icons/simple-icons/assets/20955511/b397bc66-ce77-4997-acae-24ab2306bca1)

**Comment by PratyushJayachandran** [comment]
> I made a new one following the same steps but without multiple exports and optimizations. Hopefully it is correct now.
> 
> It looks to me like it is resolved. [image](https://github.com/simple-icons/simple-icons/assets/20955511/ceb2ac66-6089-466a-a012-aa1fb9c9c799) [image](https://github.com/simple-icons/simple-icons/assets/20955511/b397bc66-ce77-4997-acae-24ab2306bca1)

Unfortunately I still see [some differences here!](https://github.com/simple-icons/simple-icons/assets/35000807/85dddaac-e1e9-4b9a-8bc1-6e56a97b9ab2)
Are you comparing the source to the SVGO optimized image?

**Comment by DenverCoder1** [comment]
I've tried doing it again but using the svgo CLI with precision set to 4 (previously I was using https://github.com/jakearchibald/svgomg which may have been causing differences?)

**Comment by PratyushJayachandran** [comment]
> I've tried doing it again but using the svgo CLI with precision set to 4 (previously I was using https://github.com/jakearchibald/svgomg which may have been causing differences?)

I don't think that was what was causing the issue. It could be precision though.. I'll compare again.

File: icons/swiprolog.svg [hunk 3]
```diff
@@ -0,0 +1 @@
...67-.5057 1.9208-.5057 1.9208s-1.0352-.7167-1.2104-1.6403c.1433.7484 1.083.9713 1.083.9713s.7321-.5378.6331-1.2518zm-14.3715.366c-.099.714.6332 1.2522.6332 1.2522s.9396-.2229 1.083-.9713c-.1752.9235-1.2102 1.64-1.2102 1.64s-.6506-1.0141-.506-1.9208zm12.5245 0c.1445.9068-.506 1.9208-.506 1.9208s-.9659-.6794-1.193-1.5678c.1876.6904 1.0658.8992 1.0658.8992s.7322-.5381.6332-1.2521zM2.5414 14.9716c-.1433.7484.637 1.3215.637 1.3215s.9392-.223 1.0826-.9713c-.1752.9395-1.226 1.6402-1.226 1.6402s-.6847-1.0668-.4936-1.9904zm18.8373 0c.191.9236-.4936 1.9904-.4936 1.9904s-1.0508-.7007-1.226-1.6402c.1433.7484 1.0826.9713 1.0826.9713s.7803-.5731.637-1.3215zm-16.9744.382c-.1433.7483.637 1.3217.637 1.3217s.8745-.208 1.0645-.895c-.2299.8863-1.2079 1.5636-1.2079 1.5636s-.6847-1.0668-.4936-1.9904zm15.1114 0c.1911.9235-.4935 1.9903-.4935 1.9903s-1.035-.7163-1.2101-1.6399c.1433.7484 1.0826.9713 1.0826.9713s.7803-.5734.621-1.3218zm-16.4011 1.704c-.1592.7483.621 1.3214.621 1.3214s.8747-.2077 1.0646-.8948c-.229.8864-1.192 1.5637-1.192 1.5637s-.6847-1.0668-.4936-1.9904zm17.7033.0696c.1444.9067-.506 1.9208-.506 1.9208s-.963-.6774-1.192-1.5638c.19.6872 1.0645.8948 1.0645.8948s.7325-.5378.6335-1.2518z"/></svg>
```
Comment (PratyushJayachandran) [inline icons/swiprolog.svg:1]:
```suggestion
<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>SWI Prolog</title><path d="M7.96 6.793a2.072 2.072 0 1 0 0 4.143 2.072 2.072 0 0 0 0-4.143zm0 3.397a1.326 1.326 0 1 1-.002-2.65 1.326 1.326 0 0 1 .001 2.65zm-.19-1.792a.397.397 0 1 1-.795 0 .397.397 0 0 1 .795 0zm9.953 2.538a2.072 2.072 0 1 0 0-4.143 2.072 2.072 0 0 0 0 4.143zm0-3.397a1.325 1.325 0 1 1 0 2.65 1.325 1.325 0 0 1 0-2.65zm-.984.86a.397.397 0 1 1 .794 0 .397.397 0 0 1-.794 0zm2.981-4.021c-.326-1.047-.718-3.103.705-4.296 0 0-2.746-.648-4.278 1.707A10.178 10.178 0 0 0 12 .907c-1.481 0-2.894.319-4.18.896C6.29-.57 3.532.082 3.532.082c1.457 1.221 1.013 3.345.683 4.366C2.381 6.49 1.253 9.25 1.253 12.29c0 2.42 1.554 5.169 1.865 8.853l1.968-1.705s.547.367 1.612.748A2.482 2.482 0 0 0 6.403 24a1.995 1.995 0 0 1-.01-.199c0-1.03.807-1.865 1.802-1.865.996 0 1.803.835 1.803 1.865l-.002.059a2.474 2.474 0 0 0 .276-2.909 18.31 18.31 0 0 0 3.255.019 2.472 2.472 0 0 0 .288 2.89c-.001-.02-.003-.039-.003-.059 0-1.03.807-1.865 1.803-1.865.996 0 1.803.835 1.803 1.865 0 .067-.004.133-.01.199a2.478 2.478 0 0 0-.234-3.767c1.153-.398 1.74-.795 1.74-.795l1.968 1.705c.312-3.684 1.865-6.434 1.865-8.853 0-3.075-1.155-5.864-3.027-7.912zm-17.683 8.55c-.147.744.63 1.316.63 1.316s.934-.233 1.081-.976c-.184.928-1.217 1.657-1.217 1.657s-.677-1.067-.494-1.996zm.543 2.055c-.147.743.63 1.315.63 1.315s.933-.233 1.08-.976c-.183.929-1.217 1.657-1.217 1.657s-.677-1.067-.493-1.996zm1.937-.367s.934-.233 1.08-.976c-.183.928-1.216 1.657-1.216 1.657s-.677-1.067-.494-1.996c-.147.743.63 1.315.63 1.315zm-.894 4.446s-.677-1.067-.493-1.996c-.147.744.63 1.316.63 1.316s.933-.234 1.08-.976c-.183.928-1.217 1.656-1.217 1.656zm1.3-1.711s-.677-1.067-.493-1.996c-.147.743.629 1.315.629 1.315s.934-.233 1.081-.976c-.184.929-1.217 1.657-1.217 1.657zm1.3-1.684s-.676-1.067-.493-1.996c-.147.744.63 1.316.63 1.316s.934-.234 1.081-.976c-.184.928-1.217 1.656-1.217 1.656zm5.717 1.07c0-1.403-1.908-3.327-2.319-4.891a4.222 4.222 0 0 1-2.542.85 4.247 4.247 0 1 1 4.247-4.246c0 .19-.017.376-.042.56a2.395 2.395 0 0 1 1.354.01 4.263 4.263 0 0 1-.043-.57 4.247 4.247 0 1 1 4.247 4.247c-.97 0-1.862-.33-2.578-.878-.395 1.57-2.324 3.507-2.324 4.918zm8.152-3.436c.184.929-.493 1.996-.493 1.996s-1.034-.729-1.217-1.657c.147.743 1.081.976 1.081.976s.776-.572.63-1.315zm-3.554.71c.148.742 1.082.976 1.082.976s.776-.572.63-1.316c.183.93-.494 1.996-.494 1.996s-1.034-.728-1.218-1.656zm1.302 1.683c.147.743 1.08.976 1.08.976s.777-.572.63-1.315c.184.929-.493 1.996-.493 1.996s-1.034-.728-1.217-1.657zm2.516 3.368s-1.033-.728-1.217-1.656c.147.742 1.082.976 1.082.976s.775-.572.629-1.316c.183.93-.494 1.996-.494 1.996zm.55-2.083s-1.033-.728-1.217-1.657c.147.743 1.081.976 1.081.976s.777-.572.63-1.315c.183.929-.494 1.996-.494 1.996zm-.675-3.711c.148.743 1.082.976 1.082.976s.776-.572.63-1.315c.183.929-.494 1.996-.494 1.996s-1.034-.729-1.218-1.657zm-6.244-1.97c0 .728-.592 1.674-1.166 2.589-.33.526-.65 1.038-.881 1.531-.232-.493-.552-1.005-.882-1.531-.573-.915-1.165-1.861-1.165-2.589A2.04 2.04 0 0 1 11.94 9.27a2.04 2.04 0 0 1 2.047 2.03z"/></svg>
```

Unfortunately I still see [some differences.](https://github.com/simple-icons/simple-icons/assets/35000807/72646c7e-eb81-42c3-aa10-b1e5625b3590)
 This suggestion should fix it but I will request someone else to review @simple-icons/icon-review

**Review by PratyushJayachandran — DISMISSED** [review]

**Comment by PratyushJayachandran** [comment]
> Most of the icon looks spot on - there's just some differences around the beak and eyes, and I think how we've generated the path.
> 
> My general rule-of-thumb is to use an _outer_ stroke of 0.5 or 1px, whereas this appears to use an inner stroke. There is no 'right' way of doing it, but just wanted to confirm with you. See my generated image overlaying your PR below.
> 
> image

image

It is cutting the eyes @adamrusted, that is why I did inner stroke instead

**Comment by PeterShaggyNoble** [comment]
Are you still working on this one, @DenverCoder1? I agree with your choice of cut-out to avoid cutting into the eyes but I'm still seeing some of the same differences mentioned above.

Also, if there's a page on their website that includes the original SVG then we should use that as `source` URL to avoid potential future link rot.

**Comment by DenverCoder1** [comment]
I'll let @PratyushJayachandran or anyone else who wants take over this issue. It seems with my current workflow, it always ends up with slight differences.

**Comment by PeterShaggyNoble** [comment]
Closing this one, so; if anyone else would like to take it up then please open a new PR for it. Thanks for your work on it so far, @DenverCoder1 👍🏻

**Pull request closed without merging**